### PR TITLE
Create order module and swagger

### PR DESCRIPTION
**Pull request by ekuritza**
Proposed changes:
This PR adds the order module and swagger documentation to the project.

Checklist:
[ ]Have you written tests for your changes?
[ ]Have you successfully run tests with your changes locally?
[x]Have you lint your code locally prior to submission?

### CODE REVIEW

File: src/order/dto/order.dto.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  status: OrderStatusEnum;
+  @IsNumber()
+  @IsNotEmpty()
+  price: number;
+  payment: {
+    cardholder: string;
+    paymentStatus: PaymentStatusEnum;
+    value: number;
+    dueDate: Date;
+  };
```
Comment (PedroFellipe) [inline src/order/dto/order.dto.ts]: order schema and dto must to match in this case, i think that the products array must to be here too

File: src/order/order-service.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import { Product } from 'src/products/entities/products';
+import { Model } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+
+@Injectable()
+export class OrderService implements OrderInterface {
+  constructor(
+    @Inject('OrderRepositoryInterface')
+    private readonly orderRepository: OrderRepositoryInterface,
+    //@InjectModel('Product') private readonly productModel: Model<Product>,
```
Comment (PedroFellipe) [inline src/order/order-service.ts]: remove this commented code, please

**Review by PedroFellipe — CHANGES_REQUESTED** [review]

**Comment by PedroFellipe** [comment]
Also, theres a lot of unit tests and integrations tests that you could to write in this PR, call us to do that bro. Especially the integrations tests will tell us that this API really works

File: src/order/dto/order.dto.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  })
+  status: OrderStatusEnum;
+  @IsNumber()
+  @IsNotEmpty()
+  price: number;
+  payment: {
+    cardholder: string;
+    paymentStatus: PaymentStatusEnum;
+    value: number;
+    dueDate: Date;
```
Comment (renatops1991) [inline src/order/dto/order.dto.ts]: I think that create a new DTO to implements this field 
```suggestion
  payment: PaymentDto
```
Please, put the correct decorators for each field
```
export class PaymentDto {
    cardholder: string;
    paymentStatus: PaymentStatusEnum;
    value: number;
    dueDate: Date;
}
```

File: src/order/entities/order.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export class Order {
+  id?: string;
+  status: OrderStatusEnum;
+  price: number;
+  payment: {
+    type: PaymentStatusEnum;
+    cardholder: string;
+    dueDate: Date;
+    value: number;
+  };
```
Comment (renatops1991) [inline src/order/entities/order.ts]: The same thing as above

File: src/order/entities/order.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  products: [
+    {
+      id: string;
+      price: number;
+      name: string;
+    },
```
Comment (renatops1991) [inline src/order/entities/order.ts]:
```suggestion
  products: ProductDto[],
```
```
export class ProductDto {
      id: string;
      price: number;
      name: string;
}
```
Please, put the correct decorators for each field

File: src/order/order-service.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import { DeleteResult } from 'mongodb';
+
+@Injectable()
+export class OrderService implements OrderInterface {
+  constructor(
+    @Inject('OrderRepositoryInterface')
+    private readonly orderRepository: OrderRepositoryInterface,
+  ) {}
+
+  async create(createOrderDto: CreateOrderDto) {
```
Comment (renatops1991) [inline src/order/order-service.ts]:
```suggestion
  async create(createOrderDto: CreateOrderDto): Promise<Order>{
```

**Review by renatops1991 — CHANGES_REQUESTED** [review]

**Pull request merged**